### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by cp-pratik-k**
### Purpose
 - add pagination in who is out card
 - add pagination in user leaves
 - add pagination in admin leaves
 - add pagination in employee repo

### Summary of Changes
 - add pagination in who is out card
 - add pagination in user leaves

### Test steps
 - run flutter test

### Conformity
- [x] Followed [git guidelines](https://github.com/canopas/flutter-developer-roadmap/blob/main/GitGuideline.md) for creating commit messages and Pull Request guidelines.
- [x] Self-approved the PR - reviewed the PR as a reviewer and gave it self-approval if everything is ok. If not, made the required changes.
- [x] Ensured that the PR satisfies all specified requirements in the ticket, including bug fixes and new features.
- [x] Provided test steps, including steps to reproduce the issue or test the new functionality, ensuring other team members can verify the changes.
- [x] Added/Updated proper code comments to make it easy-to-understand for other developers.
- [x] Reused code (if the same code was written twice, made it common and reused it at both places).
- [x] Removed unused or commented code if not required.
- [x] Ensured proper Dart naming conventions were used for variables, classes, and methods.
- [x] Localized user-facing strings.
- [x] Included screenshots/videos of behavior changes: Provided visual evidence of any changes to UI or behavior for easier review and understanding in the PR description.
- [x] Implemented proper error handling: Ensured that the code anticipated and handled potential errors and edge cases gracefully.
- [x] Avoided introducing technical debt: If the PR introduces technical debt, created and linked appropriate tickets for future resolution.
- [x] Included relevant unit tests: Wrote unit tests that focused on testing behavior and functionality, rather than merely covering lines of code.
- [x] Ensured code was performant and scalable: Verified that the changes did not introduce performance issues or bottlenecks and could scale as needed.
- [x] Ensured comments were up-to-date and relevant to the code to describe complex logic and to add understanding for other developers.
- [x] Marked the PR as ready before submitting it for review.

### CODE REVIEW

File: lib/data/services/leave_service.dart [hunk 1]
```diff
@@ -14,26 +14,76 @@ class LeaveService {
...
+  Stream<List<Leave>> userYearlyLeave(
+      {required String uid, required int year, required String spaceId}) =>
+      _leaveDb(spaceId: _userManager.currentSpaceId!)
+          .where(FireStoreConst.uid, isEqualTo: uid)
+          .where(FireStoreConst.startLeaveDate,
+          isGreaterThanOrEqualTo: DateTime(year).timeStampToInt)
+          .where(FireStoreConst.startLeaveDate,
+          isLessThanOrEqualTo: DateTime(year, 12, 31).timeStampToInt)
+          .snapshots()
+          .map((event) => event.docs.map((leave) => leave.data()).toList());
```
Comment (cp-sneha-s) [inline lib/data/services/leave_service.dart:68]: Any need for real-time fro user leave records??

File: lib/data/services/leave_service.dart [hunk 2]
```diff
@@ -14,26 +14,76 @@ class LeaveService {
```
Comment (cp-sneha-s) [inline lib/data/services/leave_service.dart:14]: Let's remove `UserStateNotifier` dependency from here

File: lib/ui/shared/who_is_out_card/bloc/who_is_out_card_bloc.dart [hunk 3]
```diff
@@ -17,56 +17,41 @@ import 'who_is_out_card_state.dart';
...
+      }
+      _subscription = getLeaveApplicationStream(
+              leaveStream:
+                  _leaveRepo.leaveByMonth(event.focusDay ?? state.focusDay),
+              membersStream: _employeeRepo.employees)
+          .listen((List<LeaveApplication> leaveApplications) {
+        add(ShowCalendarData(leaveApplications));
+      }, onError: (error, _) {
+        add(ShowCalendarError());
+      });
```
Comment (cp-sneha-s) [inline lib/ui/shared/who_is_out_card/bloc/who_is_out_card_bloc.dart:52]: we can directly emit state instead of adding event.

Comment (cp-pratik-k) [reply lib/ui/shared/who_is_out_card/bloc/who_is_out_card_bloc.dart:52]: This is not proper way to emit from stream as bloc.

File: lib/data/services/leave_service.dart (hunk 1 above)
Comment (cp-pratik-k) [reply lib/data/services/leave_service.dart:68]: no i'll remove it.

**Review by cp-sneha-s — APPROVED** [review]
